sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -425,14 +425,6 @@ send_resp_inbound_if_topic_matches(az_mqtt5_rpc_client* this_policy, az_event ev
       // read the error message if there is one
       resp_data.error_message = error_message_val_str;
     }
-    else if (
-        !az_span_is_content_equal(recv_data->payload, AZ_SPAN_EMPTY)
```

Comment:
In this scenario, content type will be AZ_SPAN_EMPTY, not random bytes, correct?

Reply:
We wouldn't know that. And we didn't check for that before either (e.g., testing if the CT is a known type).